ui/components/app/transaction-list-item/transaction-list-item.component.js

```diff
@@ -140,9 +141,18 @@ export default function TransactionListItem({
   ]);
 
   const showCancelButton = !hasCancelled && isPending && !isUnapproved;
+  let mode;
+  if (showRetryEditGasPopover) {
+    mode = EDIT_GAS_MODES.SPEED_UP;
+  } else if (showCancelEditGasPopover) {
+    mode = EDIT_GAS_MODES.CANCEL;
+  }
 
```

Comment:
Hmmm.... we only need to add wrap this in `GasFeeContextProvider` for the sake of `EditGasPopover`, right?

Also, I am realizing that wherever `EditGasPopover` is, we will need `GasFeeContextProvider`. Is there a way we can eliminate the code duplication of always wrapping `EditGasPopover` in `GasFeeContextProvider`? It would also be nice not to have to wrap components that don't need this context.

For instance, could we modify the edit-gas-popover file to export a function that wraps `EditGasPopover` with `GasFeeContextProvider`, so that the wrapping only needs to be done once?

Reply:
Hey @danjm : if we just wrap `EditGasPopover` in context it will defeat the purpose of having data available to all 3 screens in V2 flow.

There is no advantage of minimising the scope of context, a lot of times context if added to root of application like we have for i18n. Bigger scope will ensure availability of this data to other components also if needed - no downside actually.